src/opendr/perception/object_detection_2d/nanodet/algorithm/config/config_file_detail.md

```diff
@@ -0,0 +1,201 @@
+# NanoDet Config File Analysis
+
+NanoDet using [yacs](https://github.com/rbgirshick/yacs) to read yaml config file.
+
+## Saving path
+
+```yaml
+save_dir: PATH_TO_SAVE
+```
+
+Change save_dir to where you want to save logs and models. If path not exist, NanoDet will create it.
+
+## Model
+
+```yaml
+model:
+    arch:
+        name: OneStageDetector
+        backbone: xxx
+        fpn: xxx
+        head: xxx
+```
+
+Most detection model architecture can be devided into 3 parts: backbone, task head and connector between them(e.g. FPN, BiFPN, PAN...).
+
+### Backbone
+
+```yaml
+backbone:
+    name: ShuffleNetV2
+    model_size: 1.0x
+    out_stages: [2,3,4]
+    activation: LeakyReLU
+    with_last_conv: False
+```
+
+NanoDet using ShuffleNetV2 as backbone. You can modify model size, output feature levels and activation function. Moreover, NanoDet provides other lightweight backbones like **GhostNet** and **MobileNetV2**. You can also add your backbone network by importing it in `nanodet/model/backbone/__init__.py`.
+
+### FPN
+
+```yaml
+fpn:
+    name: PAN
+    in_channels: [116, 232, 464]
+    out_channels: 96
+    start_level: 0
+    num_outs: 3
+```
+
+NanoDet using modified [PAN](http://arxiv.org/abs/1803.01534) (replace downsample convs with interpolation to reduce amount of computations).
+
+`in_channels` : a list of feature map channels extracted from backbone.
+
+`out_channels` : out put feature map channel.
+
+### Head
+
+```yaml
+head:
+    name: NanoDetHead
+    num_classes: 80
+    input_channel: 96
+    feat_channels: 96
+    stacked_convs: 2
+    share_cls_reg: True
+    octave_base_scale: 8
+    scales_per_octave: 1
+    strides: [8, 16, 32]
+    reg_max: 7
+    norm_cfg:
+      type: BN
+    loss:
+```
+
+`name`: Task head class name
```

Comment:
note that this field description starts with capital letter but not the others.
There are some other occurrences like this in this file

Reply:
Thank you for your suggestion.